samples/kubernetes/helm/fhir-server/templates/deployment.yaml

```diff
@@ -0,0 +1,127 @@
+            - name: FhirServer__Security__Enabled
+              value: {{ .Values.security.enabled | quote }}
+            - name: FhirServer__Security__EnableAadSmartOnFhirProxy
+              value: {{ .Values.security.enableAadSmartOnFhirProxy | quote }}
+              {{- if .Values.security.authority }}
+            - name: FhirServer__Security__Authentication__Authority
+              value: {{ .Values.security.authority | lower | quote }}
+              {{- end }}
+              {{- if .Values.security.audience }}
+            - name: FhirServer__Security__Authentication__Audience
+              value: {{ .Values.security.audience | lower | quote }}
+              {{- end }}
+            - name: DataStore
+              value: {{ .Values.database.dataStore }}
+            {{- if eq .Values.database.dataStore "SqlServer" }}
+            - name: SqlServer__AllowDatabaseCreation
+              value: "true"
+            - name: SqlServer__Initialize
+              value: "true"
+            - name: DATABASESERVERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "fhir-server.fullname" . }}
+                  key: fullyQualifiedServerName
+            - name: DATABASEUSERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "fhir-server.fullname" . }}
+                  key: username
+            - name: DATABASEPASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "fhir-server.fullname" . }}
+                  key: password
+            - name: DATABASECATALOG
+              value: {{ include "fhir-server.fullname" . }}
+            - name: MAX_POOL_SIZE
+              value: {{ .Values.database.sql.maxPoolSize | quote }}
+            - name: SqlServer__ConnectionString
+              value: "Server=tcp:$(DATABASESERVERNAME),1433;Initial Catalog=$(DATABASECATALOG);Persist Security Info=False;User ID=$(DATABASEUSERNAME);Password=$(DATABASEPASSWORD);MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;Max Pool Size=$(MAX_POOL_SIZE)"
+            {{- end }}
+            {{- if eq .Values.database.dataStore "CosmosDb" }}
+            - name: COSMOSDBSERVER
+              value: {{ include "fhir-server.fullname" . }}
+            - name: CosmosDb__Host
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "fhir-server.fullname" . }}
+                  key: primaryEndpoint
+            - name: CosmosDb__Key
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "fhir-server.fullname" . }}
+                  key: primaryMasterKey
+            - name: FhirServer__CosmosDb__InitialCollectionThroughput
+              value: {{ .Values.database.cosmosDb.initialCollectionThroughput | quote }}
+            {{- end }}
+          ports:
+            - name: http
+              containerPort: 8080
```

Comment:
A future enhacement would be to have the option of using TLS from ingress to FHIR